piet/src/text.rs

```diff
@@ -427,6 +430,47 @@ pub trait TextLayout: Clone {
     //case would be when trimming has caused an index to not be included in the
     //layout's text?`
     fn hit_test_text_position(&self, idx: usize) -> Option<HitTestPosition>;
+
+    /// Returns a vector of `Rect`s that cover the region of the text indicated
+    /// by `range`.
+    ///
+    /// The returned rectangles are suitable for things like drawing selection
+    /// regions or highlights.
+    ///
+    /// `range` will be clamped to the length of the text if necessary.
+    fn rects_for_range(&self, range: impl RangeBounds<usize>) -> Vec<Rect> {
```

Comment:
Might be worth adding a comment that this default implementation is not BiDi capable, and that it will be removed when BiDi becomes a requirement.